File: nodejs/benchmark.js
#!/usr/bin/env node

const { fibonacci } = require('./fibonacci.js');

function benchmarkFibonacci(n) {
    const startTime = process.hrtime.bigint();
    const result = fibonacci(n);
    const endTime = process.hrtime.bigint();
    
    const executionTime = Number(endTime - startTime) / 1e9; // Convert to seconds
    return { result, executionTime };
}

function main() {
    const args = process.argv.slice(2);
    
    if (args.length !== 1) {
        console.error("Usage: node benchmark.js <n>");
        process.exit(1);
    }
    
    const n = parseInt(args[0], 10);
    if (isNaN(n)) {
        console.error("Error: Please provide a valid number");
        process.exit(1);
    }
    
    if (n < 0) {
        console.error("Error: Fibonacci is not defined for negative numbers");
        process.exit(1);
    }
    
    const { result, executionTime } = benchmarkFibonacci(n);
    console.log(`Node.js Fibonacci(${n}) = ${result}`);
    console.log(`Execution time: ${executionTime.toFixed(6)} seconds`);
}

main();

File: python/benchmark.py
#!/usr/bin/env python3

import time
import sys
from fibonacci import fibonacci


def benchmark_fibonacci(n):
    """Benchmark the fibonacci function for a given n"""
    start_time = time.perf_counter()
    result = fibonacci(n)
    end_time = time.perf_counter()
    
    execution_time = end_time - start_time
    return result, execution_time


def main():
    if len(sys.argv) != 2:
        print("Usage: python benchmark.py <n>")
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
        result, exec_time = benchmark_fibonacci(n)
        print(f"Python Fibonacci({n}) = {result}")
        print(f"Execution time: {exec_time:.6f} seconds")
    except ValueError as e:
        print(f"Error: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()